common/cloudant_base_test.go

```diff
@@ -232,4 +235,72 @@ var _ = Describe(`Cloudant custom base service UT`, func() {
 		Expect(cloudant.UserAgent).To(ContainSubstring(runtime.GOOS))
 		Expect(cloudant.UserAgent).To(ContainSubstring(runtime.GOARCH))
 	})
+
+	It("Validates cookie jar enabled for all auths", func() {
+		cloudant, err := NewBaseService(&core.ServiceOptions{
+			URL:           "https://cloudant.example",
+			Authenticator: &core.NoAuthAuthenticator{},
+		})
+		Expect(cloudant).ToNot(BeNil())
+		Expect(err).To(BeNil())
+		Expect(cloudant.BaseService.Client.Jar).ToNot(BeNil())
+
+		couchDbAuth, err := GetAuthenticatorFromEnvironment("service1")
+		Expect(err).To(BeNil())
+		cloudant, err = NewBaseService(&core.ServiceOptions{
```

Comment:
Maybe I am paranoid, but I'd create a new instance of the service each time, to get a clear page from the previous `cloudant.BaseService.Client.Jar`.